libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Skills
+{
+    public class AdaptiveSkillDialog : SkillDialog
+    {
+        [JsonProperty("$kind")]
+        public const string DeclarativeType = "Microsoft.AdaptiveSkillDialog";
+
+        [JsonConstructor]
+        public AdaptiveSkillDialog([CallerFilePath] string callerPath = "", [CallerLineNumber] int callerLine = 0)
+            : base(new SkillDialogOptions())
+        {
+            DialogOptions.Skill = new BotFrameworkSkill();
+            RegisterSourceLocation(callerPath, callerLine);
+        }
+
+        /// <summary>
+        /// Gets or sets an optional expression which if is true will disable this action.
+        /// </summary>
+        /// <example>
+        /// "user.age > 18".
+        /// </example>
+        /// <value>
+        /// A boolean expression. 
+        /// </value>
+        [JsonProperty("disabled")]
+        public BoolExpression Disabled { get; set; }
+
+        [JsonProperty("botId")]
+        public StringExpression BotId { get; set; } = "=settings.MicrosoftAppId";
+
+        [JsonProperty("skillHostEndpoint")]
+        public StringExpression SkillHostEndpoint { get; set; } = "=settings.SkillHostEndpoint";
+
+        [JsonProperty("skillAppId")]
+        public StringExpression SkillAppId { get; set; }
+
+        [JsonProperty("skillEndpoint")]
+        public StringExpression SkillEndpoint { get; set; }
+
+        /// <summary>
+        /// Gets or sets template for the activity.
+        /// </summary>
+        /// <value>
+        /// Template for the activity.
+        /// </value>
+        [JsonProperty("activity")]
```

Comment:
I am a little confused here. I saw your sample passing activity to the skill in the action, but I didn't see the activity defined in your schema. Do you mean it is optional? I want to understand how to begin a AdaptiveSkillDialog by a common message activity and a specific event activity. Can this activity just leveraging the turn.activity? 
Also, I am not sure how the activityProcessed property is used.

Reply:
@carlosscastro will send you an email describing the scenarios.

Reply:
FYI, I added activity to the schema file (I was missing that) hope it makes sense now.